Bugzilla/Template.pm

```diff
@@ -260,6 +260,129 @@ sub quoteUrls {
+    # mailto can't contain space or #, so we don't have to bother for that
+    # Do this by replacing matches with \x{FDD2}$count\x{FDD3}
+    # \x{FDDx} is used because it's unlikely to occur in the text
+    # and are reserved unicode characters. We disable warnings for now
+    # until we require Perl 5.13.9 or newer.
+    no warnings 'utf8';
+
+    # If the comment is already wrapped, we should ignore newlines when
+    # looking for matching regexps. Else we should take them into account.
+    my $s = ($comment && $comment->already_wrapped) ? qr/\s/ : qr/\h/;
+
+    # However, note that adding the title (for buglinks) can affect things
+    # In particular, attachment matches go before bug titles, so that titles
+    # with 'attachment 1' don't double match.
+    # Dupe checks go afterwards, because that uses ^ and \Z, which won't occur
+    # if it was substituted as a bug title (since that always involve leading
+    # and trailing text)
+
+    # Because of entities, it's easier (and quicker) to do this before escaping
+
+    my @things;
+    my $count = 0;
+    my $tmp;
+
+    my @hook_regexes;
+    Bugzilla::Hook::process('bug_format_comment',
+        { text => \$text, bug => $bug, regexes => \@hook_regexes,
+          comment => $comment, user => undef });
+
+    foreach my $re (@hook_regexes) {
+        my ($match, $replace) = @$re{qw(match replace)};
+        if (ref($replace) eq 'CODE') {
+            $text =~ s/$match/($things[$count++] = $replace->({matches => [
+                                                               $1, $2, $3, $4,
+                                                               $5, $6, $7, $8,
+                                                               $9, $10]}))
+                               && ("\x{FDD2}" . ($count-1) . "\x{FDD3}")/egx;
+        }
+        else {
+            $text =~ s/$match/($things[$count++] = $replace)
+                              && ("\x{FDD2}" . ($count-1) . "\x{FDD3}")/egx;
+        }
+    }
+
+    # Provide tooltips for full bug links (Bug 74355)
+    my $urlbase_re = '(' . quotemeta(Bugzilla->localconfig->{urlbase}) . ')';
+    $text =~ s~\b(${urlbase_re}\Qshow_bug.cgi?id=\E([0-9]+)(\#c([0-9]+))?)\b
+              ~($things[$count++] = get_bug_link->($3, $1, { comment_num => $5, user => $anon_user })) &&
+               ("\x{FDD2}" . ($count-1) . "\x{FDD3}")
+              ~egox;
+
+    # In the future we want to disable this in favor of Markdown's autolink.
+    # non-mailto protocols
+    my $safe_protocols = SAFE_URL_REGEXP();
+    $text =~ s~\b($safe_protocols)
+              ~($tmp = html_quote($1)) &&
+               ($things[$count++] = "<a rel=\"nofollow\" href=\"$tmp\">$tmp</a>") &&
+               ("\x{FDD2}" . ($count-1) . "\x{FDD3}")
+              ~egox;
+
+    # We have to quote now, otherwise the html itself is escaped
+    # THIS MEANS THAT A LITERAL ", <, >, ' MUST BE ESCAPED FOR A MATCH
+
+    $text = html_quote($text);
```

Comment:
HTML::Escape (which we already depend on) might be a nice change here.
it's decently fast: https://metacpan.org/pod/HTML::Escape#BENCHMARK